core/server/inventory.go

```diff
@@ -446,3 +449,16 @@ func parseInventoryFromUnstructured(obj *unstructured.Unstructured) ([]*unstruct
 
 	return objects, nil
 }
+
+func storageVersionFromHelmRelease(helmRelease *helmv2.HelmRelease) int {
+	storageVersion := helmRelease.Status.LastReleaseRevision
+
+	if storageVersion == 0 && helmRelease.Status.History != nil {
```

Comment:
🎉 

If a user is migrating from flux 2.1.2 -> 2.2.0, will they have any "residual status", e.g. should we flip this check to try for history first and then fallback to the old `LastReleaseRevision` which might still be present?

Or when the 2.2.0 controller starts it will clear `LastReleaseRevision` etc?

Reply:
https://github.com/fluxcd/helm-controller/blob/main/internal/controller/helmrelease_controller.go#L355

It explicitly clears `LastReleaseRevision`